ansible_events_ui/api/__init__.py

```diff
@@ -102,6 +102,62 @@ async def websocket_endpoint2(
                 )
                 await db.execute(query)
                 await db.commit()
+            elif data_type == "Action":
```

Comment:
The websocket handler function becomes too large and overloaded. It requires decomposition.